plugins/callback/diy.py

```diff
@@ -0,0 +1,1421 @@
+    - default_callback
+  notes:
+    - Uses the L(default,./default.html) callback plugin output when a custom callback message(C(msg)) is not provided.
+    - Makes the callback event data available via the C(ansible_callback_diy) dictionary, which can be used in the templating context for the options.
+      The dictionary is only available in the templating context for the options. It is not a variable that is available via the other
+      various execution contexts, such as playbook, play, task etc.
+    - Options being set by their respective variable input can only be set using the variable if the variable was set in a context that is available to the
+      respective callback.
+      Use the C(ansible_callback_diy) dictionary to see what is available to a callback. Additionally, C(ansible_callback_diy.top_level_var_names) will output
+      the top level variable names available to the callback.
+    - Each option value is rendered as a template before being evaluated. This allows for the dynamic usage of an option. For example,
+      C("{{ 'yellow' if ansible_callback_diy.result.is_changed else 'bright green' }}")
+    - For each respective callbacks C(msg) option, the following conditions result in the following the effects. If the option value C(is None or omit)
+      then the option is not being used, and therefore use the L(default,./default.html) callback plugin. If the option value C(is not None and not omit and
+      length is not greater than 0) then the option is being used without output, and therefore suppress output. If the option value C(is not None and not omit
+      and length is greater than 0) then the option is being used with output, and therefore output.
+    - "Valid color values: C(black), C(bright gray), C(blue), C(white), C(green), C(bright blue), C(cyan), C(bright green), C(red), C(bright cyan),
+      C(purple), C(bright red), C(yellow), C(bright purple), C(dark gray), C(bright yellow), C(magenta), C(bright magenta), C(normal)"
+  seealso:
+    - name: default – default Ansible screen output
+      description: The official documentation on the B(default) callback plugin.
+      link: ./default.html
+  requirements:
+    - set as stdout_callback in configuration
+  options:
+    on_any_msg:
+      description: Output to be used for callback on_any
+      ini:
+        - section: callback_diy
+          key: on_any_msg
+      env:
+        - name: ANSIBLE_CALLBACK_DIY_ON_ANY_MSG
+      vars:
+        - name: ansible_callback_diy_on_any_msg
+      required: False
+      type: string
+
+    on_any_msg_color:
+      description:
+        - Output color to be used for on_any_msg
+        - Template should render a L(valid color value,#notes)
+      ini:
+        - section: callback_diy
+          key: on_any_msg_color
+      env:
+        - name: ANSIBLE_CALLBACK_DIY_ON_ANY_MSG_COLOR
+      vars:
+        - name: ansible_callback_diy_on_any_msg_color
+      required: False
+      type: string
```

Comment:
```suggestion
      type: str
```
the same for the other parameters